sdk/resourcemanager/azure-resourcemanager-samples/src/main/java/com/azure/resourcemanager/compute/samples/CreateMultipleVirtualMachinesAndBatchQueryStatus.java

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.compute.samples;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.management.AzureEnvironment;
+import com.azure.core.management.Region;
+import com.azure.core.management.profile.AzureProfile;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.resourcemanager.AzureResourceManager;
+import com.azure.resourcemanager.compute.models.CachingTypes;
+import com.azure.resourcemanager.compute.models.KnownLinuxVirtualMachineImage;
+import com.azure.resourcemanager.compute.models.VirtualMachineSizeTypes;
+import com.azure.resourcemanager.resourcegraph.ResourceGraphManager;
+import com.azure.resourcemanager.resourcegraph.models.QueryRequest;
+import com.azure.resourcemanager.resources.fluentcore.utils.ResourceManagerUtils;
+import com.azure.resourcemanager.samples.Utils;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Azure Compute sample for batch create virtual machines and batch query status
+ * - Create multiple virtual machines from PIR image with data disks
+ * - Use Azure Resource Graph to batch query the status (the number of provisioningState=Succeeded ?).
+ */
+public final class CreateMultipleVirtualMachinesAndBatchQueryStatus {
+
+    /**
+     * Main function which runs the actual sample.
+     *
+     * @param azureResourceManager instance of the azure client
+     * @param resourceGraphManager instance of the azure graph client
+     * @return true if sample runs successfully
+     */
+    public static boolean runSample(AzureResourceManager azureResourceManager, ResourceGraphManager resourceGraphManager) {
+        final Long desiredVMCount = 6L;
+        final Region region = Region.US_EAST;
+        String rgName = Utils.randomResourceName(azureResourceManager, "rg-", 15);
+        List<String> vmNameList = new ArrayList<>();
+        Long total = 0L;
+        Boolean rtnValue = false;
+        try {
+            System.out.println("Creating Resource Group: " + rgName);
+            azureResourceManager.resourceGroups().define(rgName).withRegion(region).create();
+            System.out.println("Batch creating Virtual Machines");
+            for (int i = 0; i < desiredVMCount; i++) {
+                String vmName = Utils.randomResourceName(azureResourceManager, "javascvm", 15);
+                vmNameList.add("'".concat(vmName).concat("'"));
+                azureResourceManager.virtualMachines()
+                    .define(vmName)
+                    .withRegion(region)
+                    .withExistingResourceGroup(rgName)
+                    .withNewPrimaryNetwork("10.0." + i + ".0/28")
```

Comment:
For a more realistic sample, could you see if you can first create a virtual network (before the loop), and put all these VM into that virtual network (via `withExisting...`)?